internal/service/federatedsettingsidentityprovider/resource_federated_settings_identity_provider.go

```diff
@@ -114,42 +145,76 @@ func resourceMongoDBAtlasFederatedSettingsIdentityProviderRead(ctx context.Conte
 		return diag.FromErr(fmt.Errorf("error getting federated settings identity provider: %s", err))
 	}
 
-	if err := d.Set("sso_debug_enabled", federatedSettingsIdentityProvider.SsoDebugEnabled); err != nil {
-		return diag.FromErr(fmt.Errorf("error setting sso debug enabled (%s): %s", d.Id(), err))
-	}
+	if federatedSettingsIdentityProvider.GetProtocol() == "SAML" {
+		if err := d.Set("request_binding", federatedSettingsIdentityProvider.RequestBinding); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting request binding (%s): %s", d.Id(), err))
+		}
 
-	if err := d.Set("associated_domains", federatedSettingsIdentityProvider.AssociatedDomains); err != nil {
-		return diag.FromErr(fmt.Errorf("error setting associated domains list (%s): %s", d.Id(), err))
-	}
+		if err := d.Set("response_signature_algorithm", federatedSettingsIdentityProvider.ResponseSignatureAlgorithm); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting response signature algorithm (%s): %s", d.Id(), err))
+		}
 
-	if err := d.Set("okta_idp_id", federatedSettingsIdentityProvider.OktaIdpId); err != nil {
-		return diag.FromErr(fmt.Errorf("error setting OktaIdpID (%s): %s", d.Id(), err))
+		if err := d.Set("sso_debug_enabled", federatedSettingsIdentityProvider.SsoDebugEnabled); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting sso debug enabled (%s): %s", d.Id(), err))
+		}
+
+		if err := d.Set("sso_url", federatedSettingsIdentityProvider.SsoUrl); err != nil {
```

Comment:
not specific to this PR, there are many error checks like this that seem a bit verbose and we might be able to extract to some helper functions